### PR TITLE
Fix duplicate SPDX and ABIEncoderV2 lines

### PR DESCRIPTION
**Pull request by zemse**
A quick fix for #48 and #55. This fixed errors on my flattened contract that had a lot of SPDX's and a couple of ABIEncoderV2's.

Before:

```solidity
// SPDX-License-Identifier: MIT
// SPDX-License-Identifier: MIT
// SPDX-License-Identifier: GNU

pragma experimental ABIEncoderV2;
pragma experimental ABIEncoderV2;
```

After:

```solidity
// SPDX-License-Identifier: MIT AND GNU

pragma experimental ABIEncoderV2;
```

Please let me know if any changes are required!

Edit: If anyone wants to try if this PR fixes errors in your flattened contract, you can do so with the temporary package `npm install @zemse/solidity-flattener -g`.

### CODE REVIEW

**Comment by 7--** [comment]
Would be nice if this was reviewed and merged. `// SPDX-License-Identifier` always gives me issues.

**Comment by alcuadrado** [comment]
I'm really hesitant about merging this, as I suspect it leads to violating the licenses most of the time. Maybe if the identifiers aren't move around?

**Comment by zemse** [comment]
> I suspect it leads to violating the licenses most of the time.

I see what you mean. This would be very inappropriate.

I think only allowing flattening when all files have the same licenses should be fine? And throw if mixed licenses are there as commented in https://github.com/nomiclabs/truffle-flattener/issues/55#issuecomment-631815262. I feel what best could be done in case of mixed license is Axic's second point in https://github.com/ethereum/solidity/issues/8989#issuecomment-631155720, i.e. the least permissive could be picked, but I doubt if all kinds of licenses listed on https://spdx.org/licenses/ can be really arranged in an order of precedence. I'm not sure if there is any solution for mixed licenses, but for same licences is not a problem.

So to sum up, only files with same licenses would be flattened, mixed licenses would throw. Could such be acceptable?

**Comment by 7--** [comment]
Maybe make it a flag so users have to ask for is specifically like truffle-flattener file.sol --flatten-licenses. 

Or maybe throw a warning and a disclaimer if multiple licenses are detected and ask if they want to licenses to be flattened. 

Truffle-flattener shouldn't be liable if the user specifically requests it. User will remove manually if they want to. It's a lot of manual work to deploy and verify on etherscan without truffle flattener so I think it's appropriate.
